cmd/deploy/main.go

```diff
@@ -23,49 +118,42 @@ func main() {
 
 	restconfig, err := kubeconfig.ClientConfig()
 	if err != nil {
-		logger.Fatalf("Failed to initialize the kubernetes client config: %v", err)
+		return fmt.Errorf("Failed to initialize the kubernetes client config: %v", err)
 	}
 
 	client, err := kubernetes.NewForConfig(restconfig)
 	if err != nil {
-		logger.Fatalf("Failed to initialize the kubernetes clientset: %v", err)
+		return fmt.Errorf("Failed to initialize the kubernetes clientset: %v", err)
 	}
 
 	apiextClient, err := apiextclientv1beta1.NewForConfig(restconfig)
 	if err != nil {
-		logger.Fatalf("Failed to initialize the apiextensions clientset: %v", err)
+		return fmt.Errorf("Failed to initialize the apiextensions clientset: %v", err)
 	}
 
 	meteringClient, err := meteringclientv1.NewForConfig(restconfig)
 	if err != nil {
-		logger.Fatalf("Failed to initialize the metering clientset: %v", err)
+		return fmt.Errorf("Failed to initialize the metering clientset: %v", err)
 	}
 
-	deployObj, err := deploy.NewDeployer(client, apiextClient, meteringClient, logger)
+	deployObj, err = deploy.NewDeployer(deployObj, client, apiextClient, meteringClient, logger)
```

Comment:
@chancez how do you feel about passing a half-initialized `deploy.Deployer` structure to `deploy.NewDeployer`? Alternatively, could refactor the flags to initialize a variable, and not a field of the `deploy.Deployer` structure, i.e. `installCmd.Flags().StringVar(&deployObj.Namespace, ...)` could be `installCmd.Flags().StringVar(&namespace, ...)`

Reply:
Dislike passing half initialized. The latter is pretty common though.

Reply:
Maybe add a config struct instead, and pass that into the constructor.